### PR TITLE
Update docstrings for pauli measurement gate and related utilities

### PR DESCRIPTION
**Pull request by tanujkhattar**
docfix.

### CODE REVIEW

File: cirq-core/cirq/ops/measure_util.py [hunk 1]
```diff
@@ -33,19 +33,28 @@ def measure_single_paulistring(
     pauli_observable: pauli_string.PauliString,
     key: Optional[Union[str, 'cirq.MeasurementKey']] = None,
 ) -> raw_types.Operation:
-    """Returns a single PauliMeasurementGate which measures the pauli observable
+    """Returns an operation, from `cirq.PauliMeasurementGate`, to measure the `pauli_observable`.
```
Comment (dabacon) [inline cirq-core/cirq/ops/measure_util.py:36]: nit: "from" is strange here.  Maybe "a `cirq.MeasurementGate`"?

File: cirq-core/cirq/ops/measure_util.py [hunk 2]
```diff
@@ -33,19 +33,28 @@ def measure_single_paulistring(
...
+
+    Note that the coefficient of `pauli_observable` will be ignored and the resulting operation
+    will measure the single pauli observable with coefficient 1. For example:
```
Comment (dabacon) [inline cirq-core/cirq/ops/measure_util.py:39]: ugh.  Shouldn't this yell.  If I pass in `-cirq.X(q[0]) * cirq.Z(q[1])` it silently failing seems really counter intuitive.  Maybe we should open an issue on this?

File: cirq-core/cirq/ops/pauli_measurement_gate.py [hunk 3]
```diff
@@ -30,10 +30,14 @@
...
 @value.value_equality
 class PauliMeasurementGate(raw_types.Gate):
-    """A gate that measures a Pauli observable.
+    r"""A gate that measures a Pauli observable.
 
     PauliMeasurementGate contains a key used to identify results of measurement
-    and a list of Paulis which denote the observable to be measured.
+    and a list of Paulis which represent the pauli observable to be measured.
+
+    By default, a `PauliMeasurementGate` decomposes into $U M_q0 U\dagger$
```
Comment (dabacon) [inline cirq-core/cirq/ops/pauli_measurement_gate.py:38]: M_{q0}?  (here and below)

**Review by dabacon — CHANGES_REQUESTED** [review]

File: cirq-core/cirq/ops/measure_util.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/measure_util.py:36]: Yeah, I was originally planning to write:
``` 
"""Returns a `cirq.PauliMeasurementGate` to measure the `pauli_observable`.
```

but then there's the technicality that we are returning an operation and not a gate, which corresponds to applying the pauli measurement gate on a bunch of qubits. Do you have any recommendation here?

File: cirq-core/cirq/ops/measure_util.py (hunk 2 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/measure_util.py:39]: Done. https://github.com/quantumlib/Cirq/issues/5615

File: cirq-core/cirq/ops/pauli_measurement_gate.py (hunk 3 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/pauli_measurement_gate.py]: done

File: cirq-core/cirq/ops/measure_util.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/ops/measure_util.py]: Updated.

File: cirq-core/cirq/ops/measure_util.py [hunk 4]
```diff
@@ -33,20 +33,30 @@ def measure_single_paulistring(
     pauli_observable: pauli_string.PauliString,
     key: Optional[Union[str, 'cirq.MeasurementKey']] = None,
 ) -> raw_types.Operation:
-    """Returns a single PauliMeasurementGate which measures the pauli observable
+    """Returns an operation, using `cirq.PauliMeasurementGate`, to measure the `pauli_observable`.
+
+    Note that the coefficient of `pauli_observable` will be ignored and the resulting operation
```
Comment (dabacon) [inline cirq-core/cirq/ops/measure_util.py:38]: think this is out of date now with #5615 ?

**Review by dabacon — APPROVED** [review]
Think this needs to be synced with #5615 but otherwise looks good

**Comment by dabacon** [comment]
Is this out of date?

**Pull request closed without merging**